polaris-patterns/src/components/AppSettingsLayout/AppSettingsLayout.tsx

```diff
@@ -0,0 +1,77 @@
+import type {PropsWithChildren} from 'react';
+import React from 'react';
+import {
+  VerticalStack,
+  HorizontalGrid,
+  AlphaCard,
+  Box,
+  Text,
+  useBreakpoints,
+  Divider,
+} from '@shopify/polaris';
+
+export type AppSettingsLayoutProps = PropsWithChildren;
+
+export const AppSettingsLayout: React.FunctionComponent<PropsWithChildren> & {
+  AnnotatedSection: typeof AnnotatedSection;
+  Card: typeof Card;
+} = function AppSettingsLayout({children}: PropsWithChildren) {
+  return <VerticalStack gap={{xs: '8', sm: '4'}}>{children}</VerticalStack>;
+};
+
+export interface AnnotatedSectionProps {
+  title?: React.ReactNode;
+  description?: React.ReactNode;
+  id?: string;
+  hideDivider?: boolean;
+}
+
+function AnnotatedSection({
+  title,
+  description,
+  id,
+  hideDivider,
+  children,
+}: PropsWithChildren<AnnotatedSectionProps>) {
+  const {smUp} = useBreakpoints();
+  const descriptionMarkup =
+    typeof description === 'string' ? (
+      <Text as="p" variant="bodyMd">
+        {description}
+      </Text>
+    ) : (
+      description
+    );
+
+  return (
+    <>
+      {smUp && !hideDivider ? <Divider /> : null}
```

Comment:
I didn't know quite how to only render a divider between the children like we can do with css. I was thinking the options were:
1. iterate over the children in `AppSettingsLayout` and insert a `Divider` between `AnnotatedSections`. This seemed prone to errors and not flexible
2. Add a scss file and custom style to add a border on `+ .AnnotatedSection` like `LegacyCard` and `Layout.AnnotatedSection` does
3. Make the component "dumb" like I did here and add a `hideDivider` prop

LMK if you can think of another way to ignore the first instance of `AnnotatedSection` in the dom tree and conditionally render the divider

Reply:
Dumb is good, if we go with a slot approach to enforce and safeguard render order, that **might** be a good opportunity for us to relocate that concern to the AppSettingsLayout component.